pkg/pub_worker/lib/src/bin/pana_wrapper.dart

```diff
@@ -33,26 +41,174 @@ Future<void> main(List<String> args) async {
     }
   });
 
+  // Fetch the pubspec so we detect which SDK to use for analysis
+  // TODO: Download the package, extract and load the pubspec.yaml, that way
+  //       we won't have to list versions again later.
+  final pubspec = await _fetchPubspec(
+    package: package,
+    version: version,
+    pubHostedUrl: pubHostedUrl,
+  );
+
+  // Discover installed Dart and Flutter SDKs.
+  // This reads sibling folders to the Dart and Flutter SDK.
+  // TODO: Install Dart / Flutter SDKs into these folders ondemand in the future!
```

Comment:
This is a bit of a hack... we just read the SDKs from sibling directories of the default installed SDK.

I think this is a pretty cool hack :D

In the future maybe we can skip having an SDK in pub_worker at all, and instead we simply have a pre-built executable (which is faster to startup), and then we download the SDKs we need on-the-fly and cache them on the worker...

It's actually not that crazy, since we have to download the docker image containing the SDK anyways. Whether we download a large docker image or fetch an SDK tarball from GCS probably doesn't matter much.

Though if the flutter tarball is xz compressed we might need a gzipped thing :rofl:  or we'll spend a lot of time just decompressing bytes :see_no_evil: